app/charts/shared/use-chart-data.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect } from "react";
+import { useDataCubeObservationsQuery } from "../../graphql/query-hooks";
+import { Locale } from "../../locales/locales";
+import { ChartErrorType, useChartError } from "./errors";
+
+export const useChartData = ({
+  locale,
+  iri,
+  dimensions,
+  filters,
+}: {
+  locale: Locale;
+  iri: string;
+  dimensions?: string[];
+  filters?: any;
+}) => {
+  const { setChartError } = useChartError();
+  let error: ChartErrorType = "none";
+
+  const [{ data, fetching, error: dataLoadingError }] =
+    useDataCubeObservationsQuery({
+      variables: { locale, iri, dimensions, filters },
+    });
+  const observations = data?.dataCubeByIri?.observations.data;
+  const observationsPresent = observations ? observations.length > 0 : false;
+
+  if (!fetching) {
+    if (dataLoadingError) {
+      error = "dataLoading";
+    } else if (!observationsPresent) {
+      error = "noData";
+    }
+  }
+
+  useEffect(() => setChartError(error));
```

Comment:
The question is: should we clear chart errors on chart type switch and try to fetch the data again? I am not sure if it's a good approach.

Maybe it could be better to have
`useEffect(() => setChartError(error), [error, setChartError]);`
and remove `setChartError('none')` in the other place of the app, so if the data couldn't be loaded, we won't try to do this again, unless there is a page refresh. It would make the logic simpler I think, especially that other part with clearing the errors.

What do you think?